llvm/lib/Transforms/Utils/CancelCrossCUDebugInfo.cpp

```diff
@@ -0,0 +1,135 @@
+//===- CancelCrossCUDebugInfo.cpp - Enable cross cu canceling--------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Transforms/Utils/CancelCrossCUDebugInfo.h"
+#include "llvm/IR/DIBuilder.h"
+#include "llvm/IR/DebugInfoMetadata.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/InitializePasses.h"
+#include "llvm/Pass.h"
+#include "llvm/Transforms/Utils.h"
+#include "llvm/Transforms/Utils/ValueMapper.h"
+
+using namespace llvm;
+
+bool isFunctionSkipped(Function &F) {
+  return F.isDeclaration() || !F.hasExactDefinition();
+}
+
+PreservedAnalyses CancelCrossCUDebugInfo::run(Module &M,
+                                              ModuleAnalysisManager &MA) {
+
+  for (Function &F : M) {
+    if (isFunctionSkipped(F))
+      continue;
+
+    Module &M = *F.getParent();
+    DICompileUnit *cu = *M.debug_compile_units_begin();
```

Comment:
Presumably in an LTO compilation there can be multiple CompileUnits.  I don't think we always want the first?